robot/packages/testutils/fake_package_server.go

```diff
@@ -158,6 +159,14 @@ func (c *FakePackagesClientAndGCSServer) SetInvalidChecksum(flag bool) {
 	c.invalidChecksum = flag
 }
 
+// SetChecksumWithLeadingZeroes sets checksum to a valid checksum with leading zeroes.
+func (c *FakePackagesClientAndGCSServer) SetChecksumWithLeadingZeroes(flag bool) {
+	c.mu.Lock()
```

Comment:
not sure this is wrong, but out of curiosity, why lock here?

Reply:
I'm not 100% sure. I was just echoing the patterns of the other set functions above and below this one

Reply:
I think its just that this mocking server was written to be thread safe. But I feel like we don't actually use that now. The cloud_package_manager is not doing parallel downloads nor is the test. I think its fine to keep to match the convention of the rest of the file.